### DISCUSSION TITLE
Handling Delayed Web Services

### DISCUSSION THREAD

**Discussion by nealbgit** [CaffeineLab/isodata#7] [General]
CaffeineLab -

Your python-isodata package has been working great for both Public and Private/Secure data/reports.

I'm trying to figure out if the package will work with Ercot's web Market Data Transparency Web Services.  In general, they require you to start a process (e.g. export ESIIDs I own for a trade date), wait for it to complete (usually a couple of minutes), then download the resulting file.  See an example sequence below.

![image](https://github.com/user-attachments/assets/c23050fe-f5ce-4c81-8c4f-2419a83e7834)
![image](https://github.com/user-attachments/assets/b07e135e-a35d-4f15-a348-5b895188cb56)
![image](https://github.com/user-attachments/assets/933acc13-ca05-4ccb-92ec-c10dd97320ca)
![image](https://github.com/user-attachments/assets/1d57d604-b5c8-44de-95b7-cf781a2b7e93)

In the last slide, after the export file is ready, the 'INITIALIZED' text turns to a computer icon (like the items below it), which you can then click for the file download.

I could probably use Selenium for a web-scrape, with some waits and retries, but if I were able to access the data via API (and your package), it would be much cleaner.

Please let me know your thoughts.

**Comment by CaffeineLab** [comment]
I did a quick search for anything that would help automate this for you and didn't come across anything.  When I have some free time later this week I may look again, but since it's not in the EIP document or the .wsdl I am not too hopeful.  If you could schedule it, then you could expect it to be in the ERCOT Extracts and then just synch that folder via the info request service.... That's my initial thought anyways...  

Glenn

**Reply by nealbgit** [reply]
Thanks Glenn.  Will take a look at scheduling it.  But if you discover anything that might simplify it, please let me know!

**Comment by CaffeineLab** [comment]
Same - if you turn over some new info let me know and I'll gladly get it incorporated.
